src/LibHdb++Defines.h

```diff
@@ -0,0 +1,173 @@
+/* Copyright (C) : 2014-2017
+   European Synchrotron Radiation Facility
+   BP 220, Grenoble 38043, FRANCE
+
+   This file is part of libhdb++cassandra.
+
+   libhdb++cassandra is free software: you can redistribute it and/or modify
+   it under the terms of the Lesser GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   libhdb++cassandra is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the Lesser
+   GNU General Public License for more details.
+
+   You should have received a copy of the Lesser GNU General Public License
+   along with libhdb++cassandra.  If not, see <http://www.gnu.org/licenses/>. */
+
+#ifndef _HDBPP_LIBHDBPP_DEFINES_H
+#define _HDBPP_LIBHDBPP_DEFINES_H
+
+#include <string>
+
+/** 
+ * @namespace HDBPP
+ * @brief HDBPP contains all the various classes required for the libhdbpp-cassandra 
+ * shared library
+ *
+ * The namaspace currently contains both the exposed HdbPPCassandra class and a number 
+ * of helper classes that break out functionality from HdbPPCassandra. In future it 
+ * may be the case we hide some of these helper classes inside a inner namespace,
+ * to unclutter the HDBPP namespace.
+ */
+
+/**
+ * @defgroup HDBPP-Interface HDBPP Interface
+ * External interface used to archive tango event data into cassandra.
+ */
+
+/**
+ * @defgroup HDBPP-Implementation HDBPP Implementation
+ * Internal implementation classes for this shared library
+ */
+

```

Comment:
Comments above are just doxygen tags to define some groups etc.

Yes, moved to stop polluting the HDBPP namespace. Moved to HDBPPInner as discussed,